dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehavior.cs

```diff
@@ -30,6 +31,32 @@ public static FunctionChoiceBehavior Auto(IEnumerable<KernelFunction>? functions
         return new AutoFunctionChoiceBehavior(functions, autoInvoke);
     }
 
+    /// <summary>
+    /// Gets an instance of the <see cref="FunctionChoiceBehavior"/> that provides either all of the <see cref="Kernel"/>'s plugins' functions to the AI model to call or specific ones.
+    /// This behavior forces the model to always call one or more functions. The model will then select which function(s) to call.
+    /// </summary>
+    /// <param name="functions">
+    /// Functions to provide to the model. If null, all of the <see cref="Kernel"/>'s plugins' functions are provided to the model.
+    /// If empty, no functions are provided to the model, which is equivalent to disabling function calling.
+    /// </param>
+    /// <param name="autoInvoke">
+    /// Indicates whether the functions should be automatically invoked by AI connectors.
+    /// </param>
+    /// <param name="functionsSelector">
+    /// The callback function allows customization of function selection.
+    /// It accepts functions, chat history, and an optional kernel, and returns a list of functions to be used by the AI model.
+    /// This should be supplied to prevent the AI model from repeatedly calling functions even when the prompt has already been answered.
+    /// For example, if the AI model is prompted to calculate the sum of two numbers, 2 and 3, and is provided with the 'Add' function,
+    /// the model will keep calling the 'Add' function even if the sum - 5 - is already calculated, until the 'Add' function is no longer provided to the model.
+    /// In this example, the function selector can analyze chat history and decide not to advertise the 'Add' function anymore.
```

Comment:
The behavior described here is just one of many that would need unique logic to handle it. I don't think we should default to the example scenario above because it will be replaced 100% of the time when a custom scenario is modeled.